### PR TITLE
-Xlint:serial,implicit-not-found

### PR DESCRIPTION
**Pull request by som-snytt**
Promote linting implicitNotFound and
implicitAmbiguous to -Xlint. Also put
warnings about serialVersionUID behind -Xlint.

### CODE REVIEW

**Comment by som-snytt** [comment]
This would get the build a baby step closer to lintable.

You especially want to be able to turn down messaging that is incorrect.

```
[warn] /home/amarki/projects/scala/src/library/scala/Enumeration.scala:303: Invalid implicitNotFound message for value ev:
[warn] The type parameter B referenced in the message of the @implicitNotFound annotation is not defined by value ev.
[warn]     override def zip[B](that: Iterable[B])(implicit @implicitNotFound(ValueSet.zipOrdMsg) ev: Ordering[(Value, B)]): SortedIterableCC[(Value, B)] =
[warn]                                                                                           ^

```

File: src/compiler/scala/tools/nsc/settings/Warnings.scala [hunk 1]
```diff
@@ -98,6 +98,8 @@ trait Warnings {
     val Constant               = LintWarning("constant",                  "Evaluation of a constant arithmetic expression results in an error.")
     val Unused                 = LintWarning("unused",                    "Enable -Ywarn-unused:imports,privates,locals,implicits.")
     val NonlocalReturn         = LintWarning("nonlocal-return",           "A return statement used an exception for flow control.")
+    val ImplicitNotFound       = LintWarning("implicit-not-found",        "Check @implicitNotFound and @implicitAmbiguous messages.")
+    val Serial                 = LintWarning("serial",                    "@SerialVersionUID on traits and non-serializable classes.")
```
Comment (hrhino) [inline src/compiler/scala/tools/nsc/settings/Warnings.scala:102]: There's something bittersweet about getting your warning put behind a lint flag. Far fewer people will see it, but it gets to be mentioned in more `.sbt` files. (Hopefully not preceded with a `-`.)

**Review by hrhino — APPROVED** [review]

File: src/compiler/scala/tools/nsc/settings/Warnings.scala (hunk 1 above)
Comment (SethTisue) [reply src/compiler/scala/tools/nsc/settings/Warnings.scala:102]: if your warning ends up in @tpolecat's blog post, you can retire happy.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/settings/Warnings.scala:102]: if you can't get tpolecat mad at you, you're not doing your job.

**Comment by som-snytt** [comment]
Rebased. `soylint` as in `soylent`. If there are no lint warnings, the reporter could add:
```
Soylint green!
```

**Comment by som-snytt** [comment]
In the same vein, why are files I scrubbed with a toothbrush five years ago suddenly covered with a grimy patina? I guess the [flags file](https://github.com/scala/scala/commit/48d14aaaced95ef2577bf8bc799c4a74788e4663) was never needed.

**Pull request merged**